vignettes/sir_models.Rmd

```diff
@@ -0,0 +1,680 @@
+transmission <- contact$matrix /
+  rep(contact$demography$population, each = ncol(contact$matrix))
+transmission
+```
+
+This can be given as a parameter to the model by adding the argument `data = list(m = transmission)` when using the `new()` method on the `gen_age` generator.
+
+## Saving a model into a package
+
+If you want to distribute your model in an R package, rather than building it locally, you will want to use the `odin_dust_package()` function instead. This will write the transpiled C++ dust code into `src` and its R interface in `R/dust.R`. Package users are not required to regenerate the dust code, and their compiler will build the library when they install the package. You may also wish to add a file `R/zzz.R` to your package `mycoolpackage` with the following lines:
+
+```r
+##' @useDynLib mycoolpackage, .registration = TRUE
+NULL
+```
+
+to help automate the compilation of the model.
+
+# Running the SIR model with [`dust`](https://mrc-ide.github.io/dust/)
+
+Now we can use the `new` method on the generator to make `dust` objects. Typically you will work through the `mcstate` package to run the model, but for illustrative purposes we show here that `dust` can be driven directly from R.
+
+[`new()`](https://mrc-ide.github.io/dust/reference/dust.html#method-new) takes the data needed to run the model i.e. a list with any parameters defined as `user` in the odin code above, the value of the initial step $t_0$, and the number of particles, each for now can simply be thought of as an independent stochastic realisation of the model, but in the next step will be used when inferring model parameters.
+
+Additional arguments include the number of threads to parallelise the particles over, and the seed for the random number generator. The seed must be an integer, and using the same seed will ensure reproducible results for all particles. For threads to take effect, `openmp` must be enabled in the compilation step, which you can check by running:
+
+```{r}
+gen_sir$public_methods$has_openmp()
+```
+
+To use this to directly create a new dust object with 10 particles, run using 4 threads:
+
+```{r}
+sir_model <- gen_sir$new(data = list(dt = 1, S_ini = 1000, I_ini = 10, beta = 0.2, gamma = 0.1),
+                         step = 1,
+                         n_particles = 10L,
+                         n_threads = 4L,
+                         seed = 1L)
+```
+
+Check the initial state is ten particles wide, with four states deep ($t$, $S$, $I$, $R$):
+
+```{r}
+sir_model$state()
+```
+
+Run the particles (repeats) forward 10 steps of length $dt$, followed by another 10 steps:
+
+```{r}
+sir_model$run(10)
+sir_model$run(20)
+```
+
+To change the parameters, the model does not need to be recompiled. We simply use the generator object to make a new model. Let's do this and increase the infection rate, and the population size. We will also use a smaller time step to calculate multiple transitions per unit time:
```

Comment:
I think that the compilation part here is a red-herring (not the detail the user will be expecting). You might write this as

```suggestion
We can change the parameters, say by increasing the infection rate and the population size, by reinitalising the model. We will also use a smaller time step to calculate multiple transitions per unit time:
```

**However** this goes against advice in the dust vignette to use `$reset()` if you want to change parameters, so you might do this instead with that function